pkg/cmd/openshift-network-controller/cmd.go

```diff
@@ -0,0 +1,166 @@
+	"github.com/openshift/origin/pkg/cmd/server/apis/config/validation"
+	"github.com/openshift/origin/pkg/configconversion"
+)
+
+const RecommendedStartNetworkControllerName = "openshift-network-controller"
+
+type OpenShiftNetworkController struct {
+	ConfigFilePath string
+	Output         io.Writer
+}
+
+var longDescription = templates.LongDesc(`
+	Start the OpenShift SDN controller`)
+
+func NewOpenShiftNetworkControllerCommand(name, basename string, out, errout io.Writer) *cobra.Command {
+	options := &OpenShiftNetworkController{Output: out}
+
+	cmd := &cobra.Command{
+		Use:   name,
+		Short: "Start the OpenShift SDN controller",
+		Long:  longDescription,
+		Run: func(c *cobra.Command, args []string) {
+			kcmdutil.CheckErr(options.Validate())
+
+			serviceability.StartProfiler()
+
+			if err := options.StartNetworkController(); err != nil {
+				if kerrors.IsInvalid(err) {
+					if details := err.(*kerrors.StatusError).ErrStatus.Details; details != nil {
+						fmt.Fprintf(errout, "Invalid %s %s\n", details.Kind, details.Name)
+						for _, cause := range details.Causes {
+							fmt.Fprintf(errout, "  %s: %s\n", cause.Field, cause.Message)
+						}
+						os.Exit(255)
+					}
+				}
+				glog.Fatal(err)
+			}
+		},
+	}
+
+	flags := cmd.Flags()
+	// This command only supports reading from config
+	flags.StringVar(&options.ConfigFilePath, "config", options.ConfigFilePath, "Location of the master configuration file to run from.")
+	cmd.MarkFlagFilename("config", "yaml", "yml")
+	cmd.MarkFlagRequired("config")
+
+	return cmd
+}
+
+func (o *OpenShiftNetworkController) Validate() error {
+	if len(o.ConfigFilePath) == 0 {
+		return errors.New("--config is required for this command")
+	}
+
+	return nil
+}
+
+// StartNetworkController calls RunNetworkController and then waits forever
+func (o *OpenShiftNetworkController) StartNetworkController() error {
+	if err := o.RunNetworkController(); err != nil {
+		return err
+	}
+
+	go daemon.SdNotify(false, "READY=1")
+	select {}
+}
+
+// RunNetworkController takes the options and starts the network controller
+func (o *OpenShiftNetworkController) RunNetworkController() error {
+	// try to decode into our new types first.  right now there is no validation, no file path resolution.  this unsticks the operator to start.
+	// TODO add those things
+	configContent, err := ioutil.ReadFile(o.ConfigFilePath)
+	if err != nil {
+		return err
+	}
+	scheme := runtime.NewScheme()
+	utilruntime.Must(openshiftcontrolplanev1.Install(scheme))
+	codecs := serializer.NewCodecFactory(scheme)
+	obj, err := runtime.Decode(codecs.UniversalDecoder(openshiftcontrolplanev1.GroupVersion, configv1.GroupVersion), configContent)
+	if err == nil {
+		// Resolve relative to CWD
+		absoluteConfigFile, err := api.MakeAbs(o.ConfigFilePath, "")
+		if err != nil {
+			return err
+		}
+		configFileLocation := path.Dir(absoluteConfigFile)
+
+		config := obj.(*openshiftcontrolplanev1.OpenShiftControllerManagerConfig)
```

Comment:
please open a high severity bugzilla to tighten this config.  We can't/won't ship it like this because the configuration needs to be tighter.